platforms/core-execution/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/WatchableHierarchies.java

```diff
@@ -277,19 +276,6 @@ public Stream<File> stream() {
         return hierarchies.stream();
     }
 
-    private void checkThatNothingExistsInNewWatchableHierarchy(String watchableHierarchy, SnapshotHierarchy vfsRoot) {
-        vfsRoot.rootSnapshotsUnder(watchableHierarchy)
-            .filter(snapshotRoot -> !isInWatchableHierarchy(snapshotRoot.getAbsolutePath()) && !ignoredForWatching(snapshotRoot))
-            .findAny()
-            .ifPresent(snapshotRoot -> {
-                throw new IllegalStateException(String.format(
-                    "Found existing snapshot at '%s' for unwatched hierarchy '%s'",
-                    snapshotRoot.getAbsolutePath(),
-                    watchableHierarchy
-                ));
-            });
-    }
-
```

Comment:
I don't think we should do this. There is an invariant held currently that whatever we start watching should not have known content from before. If this is violated by CC, we should fix that instead. Why does CC violate this?

Reply:
Isn't this behavior the result of the changes here in 73f070fad599c821fdfebb89ad3b6bf767d893c2?